lib/yang_wrappers.c

```diff
@@ -1138,6 +1139,30 @@ void yang_str2mac(const char *value, struct ethaddr *mac)
 	(void)prefix_str2mac(value, mac);
 }
 
+struct yang_data *yang_data_new_date_and_time(const char *xpath, time_t time)
+{
+	struct tm tm;
+	char timebuf[MONOTIME_STRLEN];
+	struct timeval _time, time_real;
+	char *ts_dot;
+
+	_time.tv_sec = time;
+	_time.tv_usec = 0;
+	monotime_to_realtime(&_time, &time_real);
+
+	gmtime_r(&time_real.tv_sec, &tm);
+
+	/* rfc-3339 format */
+	strftime(timebuf, sizeof(timebuf), "%Y-%m-%dT%H:%M:%S", &tm);
+	ts_dot = timebuf + strlen(timebuf);
+
+	/* microseconds and appends Z */
+	snprintfrr(ts_dot, sizeof(timebuf) - strlen(timebuf), ".%06luZ",
+		   (unsigned long)time_real.tv_usec);
```

Comment:
this technically works, but it would be better to `snprintf` into a separate buffer, use strlcat and check the return value to make sure; mixing pointer math and printf's should be avoided where possible